st2common/st2common/services/action.py

```diff
@@ -97,10 +101,22 @@ def request(liveaction):
     execution = executions.create_execution_object(liveaction, publish=False)
 
     # Update or create trace before publishing the execution
-    _, trace_db = trace_service.get_trace_db_by_live_action(liveaction=liveaction)
-    trace_service.add_or_update_given_trace_db(
-        trace_db=trace_db,
-        action_executions=[str(execution.id)])
+    if getattr(liveaction, 'context', None) and TRACE_CONTEXT in liveaction['context']:
+        trace_context = liveaction['context'][TRACE_CONTEXT]
+        try:
+            _, trace_db = trace_service.get_trace_db_by_live_action(liveaction=liveaction)
+        except StackStormDBObjectNotFoundError:
+            msg = 'No trace object found in db for context %s.' % trace_context
+            LOG.exception(msg)
+            raise UniqueTraceNotFoundException(msg)
```

Comment:
Not sure why we want to raise a UniqueTraceNotFoundException here, the StackStormDBObjectNotFoundError seems reasonable.